http/server.go

```diff
@@ -152,26 +112,45 @@ func (server *APIServer) StartSecured(edgeMode bool) error {
 
 	httpServer := &http.Server{
 		Addr:         listenAddr,
-		Handler:      h,
+		Handler:      server.nonEdgeHandler(httpHandler),
 		ReadTimeout:  120 * time.Second,
 		TLSConfig:    tlsConfig,
 		WriteTimeout: 30 * time.Minute,
 	}
-	
+
 	go func() {
 		securityShutdown := config.AgentOptions.AgentSecurityShutdown
 		time.Sleep(securityShutdown)
 
 		if !server.signatureService.IsAssociated() {
-			log.Printf("[INFO] [main,http] [message: Shutting down API server as no client was associated after %s, keeping alive to prevent restart by docker/kubernetes]", securityShutdown)
+			log.Printf("[INFO] [http] [message: Shutting down API server as no client was associated after %s, keeping alive to prevent restart by docker/kubernetes]", securityShutdown)
 
 			err := httpServer.Shutdown(context.Background())
 			if err != nil {
-				log.Fatalf("[ERROR] [server] [message: failed shutting down server] [error: %s]", err)
+				log.Fatalf("[ERROR] [http] [message: failed shutting down server] [error: %s]", err)
 			}
 
 		}
 	}()
 
 	return httpServer.ListenAndServeTLS(agent.TLSCertPath, agent.TLSKeyPath)
 }
+
+func (server *APIServer) edgeHandler(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if !server.edgeManager.IsKeySet() {
+			httpError.WriteError(w, http.StatusForbidden, "Unable to use the unsecured agent API without Edge key", errors.New("edge key not set"))
+			return
+		}
+
+		server.edgeManager.ResetActivityTimer()
+
+		next.ServeHTTP(w, r)
+	})
+}
+
+func (server *APIServer) nonEdgeHandler(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
isn't this whole function the same as `return next`? why do we need this?

Reply:
yes you're right, I think it can be simplified by just deleting the function